components/layout_thread/lib.rs

```diff
@@ -1153,11 +1153,8 @@ impl LayoutThread {
             shared_layout_context.style_context.stylist.rule_tree.dump_stdout();
         }
 
-        // GC The rule tree.
-        //
-        // FIXME(emilio): The whole point of the free list is not always freeing
-        // the list, find a good heuristic here for that.
-        unsafe { shared_layout_context.style_context.stylist.rule_tree.gc() }
+        // GC the rule tree if some heuristics are met.
+        unsafe { shared_layout_context.style_context.stylist.rule_tree.maybe_gc(); }
```

Comment:
I guess this will need the patch over in https://bugzilla.mozilla.org/show_bug.cgi?id=1318238 to avoid leaking right? Feel free to drop it here :)